pkg/util/azureclient/mgmt/resources/deployments_addons.go

```diff
@@ -15,6 +15,7 @@ import (
 type DeploymentsClientAddons interface {
 	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) error
 	Wait(ctx context.Context, resourceGroupName string, deploymentName string) error
+	GetDeployment(ctx context.Context, resourceGroupName, deploymentName string) (resources.DeploymentExtended, error)
```

Comment:
Add `Get(ctx context.Context, resourceGroupName, deploymentName string) (resources.DeploymentExtended, error)` to the DeploymentsClient interface instead